app/src/main/java/org/mozilla/tv/firefox/IntentValidator.kt

```diff
@@ -57,10 +60,18 @@ object IntentValidator {
 
         when (intent.action) {
             Intent.ACTION_MAIN -> {
-                val dialParams = intent.extras?.getString(DIAL_PARAMS_KEY) ?: return null
-                if (dialParams.isNotEmpty()) {
+                val extras = intent.extras ?: return null
+
+                if (extras.keySet().contains(DIAL_PARAMS_KEY)) {
+                    val param = extras.getString(DIAL_PARAMS_KEY)
                     TelemetryIntegration.INSTANCE.youtubeCastEvent()
-                    return ValidatedIntentData(url = "https://www.youtube.com/tv?$dialParams", source = Session.Source.ACTION_VIEW)
+                    return ValidatedIntentData(url = "https://www.youtube.com/tv?$param", source = Session.Source.ACTION_VIEW)
+                }
+                // The fetch delay can be changed for testing purposes through an intent with custom flags
+                // isQA and fetchDelay. Where isQA is true/false and fetchDelay is the intended next fetch time
+                // in seconds.
+                else if (extras.keySet().contains(IS_QA_BUILD_KEY)) {
```

Comment:
This is not mutually exclusive with `DIAL_PARAMS_KEY`: does this have to be an `else if`? Since we may return from the block above, I think this should actually be before that block.

fwiw, there's a saying, "no else after return", that I never really understood. However, it's redundant to return in a previous block and then add, "else if": maybe that's why?